de.dlr.sc.virsat.project.ui/src/de/dlr/sc/virsat/project/ui/navigator/dropAssist/DVLMObjectDropAdapterAssistant.java

```diff
@@ -9,98 +9,176 @@
  *******************************************************************************/
 package de.dlr.sc.virsat.project.ui.navigator.dropAssist;
 
-import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
 
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IConfigurationElement;
 import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
 import org.eclipse.core.runtime.Status;
-import org.eclipse.emf.common.command.Command;
-import org.eclipse.emf.ecore.EObject;
-import org.eclipse.jface.util.LocalSelectionTransfer;
-import org.eclipse.jface.viewers.ISelection;
 import org.eclipse.jface.viewers.IStructuredSelection;
-import org.eclipse.jface.viewers.StructuredSelection;
-import org.eclipse.swt.dnd.DND;
 import org.eclipse.swt.dnd.DropTargetEvent;
 import org.eclipse.swt.dnd.TransferData;
-import org.eclipse.ui.PlatformUI;
 import org.eclipse.ui.navigator.CommonDropAdapter;
 import org.eclipse.ui.navigator.CommonDropAdapterAssistant;
-import org.eclipse.ui.navigator.CommonViewer;
-
-import de.dlr.sc.virsat.project.editingDomain.VirSatEditingDomainRegistry;
-import de.dlr.sc.virsat.project.editingDomain.VirSatTransactionalEditingDomain;
-import de.dlr.sc.virsat.project.editingDomain.commands.VirSatEditingDomainClipBoard.ClipboardState;
-import de.dlr.sc.virsat.project.editingDomain.commands.VirSatPasteFromClipboardCommand;
-import de.dlr.sc.virsat.project.ui.navigator.VirSatNavigator;
+import de.dlr.sc.virsat.model.dvlm.categories.ATypeInstance;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.dvlm.concepts.util.ActiveConceptHelper;
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralElementInstance;
+import de.dlr.sc.virsat.project.Activator;
 
 /**
- * VirSat Drop Assitant to copy and paste SEIs and CAs in the Navigator
- * @author fisc_ph
- *
+ * Drop Assistant for VirSat Navigator that delegates calls to concept specific drop assistants
  */
 public class DVLMObjectDropAdapterAssistant extends CommonDropAdapterAssistant {

```

Comment:
> This shoud be named somehow like DelegatedDropAdapterAssistant

Done.

Reply:
> To me it looks like this class mostly repeats platform code. Can't we just use the normal navigator dropassistant extension point from concepts? We can make them higher priority to be sure they act before the default ones

Unfortunately no. From what i saw, we can only add dropAdapter to navigator content contributions. Therefore every concept may have to provide a complete navigator content. Not a 100% sure, but i also think that it is only possible to add one dropAdapter per navigatorContent.

The way it is implemented now, it is possible to define a dropAdapter which can be used in any navigator but it can also be reused in the delegateDropAdapter.

I think this increases functionality and flexibility on what developers want to do on concept side.